package.json

```diff
@@ -62,6 +62,7 @@
     "nock": "^13.2.4",
     "pre-commit": "^1.2.2",
     "proxy": "^1.0.2",
+    "ramda": "^0.28.0",
```

Comment:
Do we really need to introduce a new dependency? Can't we implemented what we need or is it too expensive?

You're using concat and a deep merge function, however the schema we have to merge does not feel that deep so it might be feasible?

Reply:
As you wish, I added this dependencies because they asked me to do so, but I wrote also a version without additional dependencies

Reply:
I just remembered why we implemented the merge with the `allOf` property, it's because if we have two schemas like:
```json
{
  "if": {
    "properties": {
      "field": {
        "const": "value"
      }
    }
  },
  "then": {
    "required": [
      "field1"
    ]
  }
}
```
and
```json
{
  "if": {
    "properties": {
      "anotherField": {
        "const": "anotherValue"
      }
    }
  },
  "then": {
    "required": [
      "anotherField1"
    ]
  }
}
```

a deep merge would result in something like this:
```json
{
{
  "if": {
    "properties": {
      "anotherField": {
        "const": "anotherValue"
      },
      "field": {
        "const": "value"
      }
    }
  },
  "then": {
    "required": [
      "field2",
      "anotherField1"
    ]
  }
}
```
and a shallow merge in something like this:
```json
{
  "if": {
    "properties": {
      "anotherField": {
        "const": "anotherValue"
      }
    }
  },
  "then": {
    "required": [
      "anotherField1"
    ]
  }
}
```

both results are wrong in that case.

Now in our default schema we don't have complicated properties like`if` or `$ref` and stuff like that so we might be able to ignore this problem, but if we do that I would add a note on the default schema warning that it must not have other properties than `properties, required and additionalProperties` to make sure we don't have unexpected behaviours.

what do you guys think?

Reply:
In my opinion it's ok, sinche the baseEnv is a costant. For this part I want to know if you prefer me to add the rambda dependency or use the code I wrote in the first version

Reply:
You can go with the previous version if you want :+1:

Reply:
I removed the dependency.
For the warning part, where would you like to add the note about the object merging?

Reply:
right now the function has this very generic name `mergeJsonSchemas`, this can be also misleading so I would rename it to `mergeWithDefaultJsonSchema` or something like that so nobody uses it for any other reason, and with that I would bring the default json schema inside the function instead of going as a parameter. In all of that I would put the warning note right above the declaration of the default json schema like:
```javascript
// Warning: including any first level properties other than the ones already declared may have undesired effects on the result of the merge
const defaultSchema = {
...
}
```

Reply:
for the warning let's also add a test case for documentation

Reply:
yes, exactly: this way the "wrong behaviour" is well documented and already have the test case for when we want to fix it